src/wallet.js

```diff
@@ -279,10 +282,32 @@ const wallet = {
    * @memberof Wallet
    * @inner
    */
-  hashPassword(password) {
+  oldHashPassword(password) {
     return CryptoJS.SHA256(CryptoJS.SHA256(password));
   },
 
+  /**
+   * Get the hash of a password (hmac + salt)
+   *
+   * @param {string} password Password to be hashes
+   * @param {string} salt Salt to be used when hashing the password. If not passed, we generate one
+   *
+   * @return {Object} Object with {key, salt}
+   *
+   * @memberof Wallet
+   * @inner
+   */
+  hashPassword(password, salt) {
+    if (salt === undefined) {
+      salt = CryptoJS.lib.WordArray.random(128 / 8).toString();
+    }
+    const key = CryptoJS.PBKDF2(password, salt, {
```

Comment:
As far as I found, PBKDF2 of CryptoJS uses SHA1 by default [1]. We should document it as well.

[1] https://github.com/sytelus/CryptoJS/blob/master/components/pbkdf2.js#L30

Reply:
Done!